storage/src/vespa/storage/common/message_guard.h

```diff
@@ -0,0 +1,39 @@
+// Copyright Yahoo. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+#pragma once
+
+#include "messagesender.h"
+#include <memory>
+#include <mutex>
+#include <vector>
+
+namespace storage {
+
+class MessageGuard {
```

Comment:
Will consider for a follow-up. This is just old code that was moved up and out a level since it was used by both the distributor and the merge throttler, but for some reason was placed under the distributor sub-library 🤔